src/pages/billing/_tests/CreateAzureBillingProjectModal.test.js

```diff
@@ -0,0 +1,239 @@
+    // Assert
+    await screen.findByText(noManagedApps)
+    expect(screen.queryByText(invalidUuidError)).toBeNull()
+    verifyDisabled(getManagedAppInput())
+    verifyDisabled(getCreateButton())
+  })
+
+  it('shows an error if the listAzureManagedApplications Ajax call errors', async () => {
+    // Arrange - Mock managed app Ajax call to return a server error.
+    Ajax.mockImplementation(() => {
+      return {
+        Billing: { listAzureManagedApplications: () => Promise.reject('expected test failure-- ignore console.error message') }
+      }
+    })
+    // Act - Supply valid UUID
+    await act(async () => {
+      await userEvent.type(getSubscriptionInput(), uuid())
+    })
+    // Assert
+    await screen.findByText(managedAppCallFailed)
+    expect(screen.queryByText(invalidUuidError)).toBeNull()
+    verifyDisabled(getManagedAppInput())
+    verifyDisabled(getCreateButton())
+  })
+
+  it('renders available managed applications and can create a project', async () => {
+    // Arrange - Mock managed app and create project Ajax calls to succeed.
+    const createResult = {}
+    const projectName = 'Billing_Project_Name'
+    const appName = 'appName'
+    const tenant = 'tenant'
+    const subscription = 'subscription'
+    const mrg = 'mrg'
+    Ajax.mockImplementation(() => {
+      return {
+        Billing: {
+          listAzureManagedApplications: () => Promise.resolve(
+            {
+              managedApps: [
+                { applicationDeploymentName: 'testApp1', tenantId: 'fakeTenant1', subscriptionId: 'fakeSub1', managedResourceGroupId: 'fakeMrg1' },
+                { applicationDeploymentName: appName, tenantId: tenant, subscriptionId: subscription, managedResourceGroupId: mrg }
+              ]
+            }
+          ),
+          createAzureProject: (billingProjectName, tenantId, subscriptionId, managedResourceGroupId) => {
+            createResult.billingProjectName = billingProjectName
+            createResult.tenantId = tenantId
+            createResult.subscriptionId = subscriptionId
+            createResult.managedResourceGroupId = managedResourceGroupId
+            Promise.resolve(createResult)
+          }
+        }
+      }
+    })
+
+    // Act - Insert valid billing project name.
+    await act(async () => {
+      await userEvent.type(getBillingProjectInput(), projectName)
+    })
+    // Assert
+    verifyDisabled(getCreateButton())
+
+    // Act - Supply valid subscription UUID and wait for Ajax response
+    await act(async () => {
+      await userEvent.type(getSubscriptionInput(), uuid())
+    })
+    await waitFor(() => verifyEnabled(getManagedAppInput()))
+    // Assert
+    verifyDisabled(getCreateButton())
+
+    // Act - Select one of the managed apps
+    await userEvent.click(getManagedAppInput())
+    const selectOption = await screen.findByText(appName)
+    await userEvent.click(selectOption)
+    // Assert
+    verifyEnabled(getCreateButton())
+
+    // Act - Click Create
```

Comment:
I don't see how this adds value to the code, which already says this.

Reply:
I tend to agree, but @msilva-broad and I talked and there is a strong desire to make the tests follow a standardized pattern that will make it easier for others to skim them and understand what is going on. So for the purposes of standardization/education, I'm OK with it being slightly more verbose.